### PR TITLE
Fine tuning of error_handling debug tools

### PR DESCRIPTION
**Pull request by hoodmane**
I fixed a typo in `EM_JS_REF` and `EM_JS_NUM`: namely `LOG_EM_JS_ERROR(func_name, err);` ==> `LOG_EM_JS_ERROR(func_name, e);`

Also added an extra function `log_error_obj` which takes a `JsRef` and does `console.error(jsobj)`, useful for debugging.

### CODE REVIEW

File: src/core/error_handling.h [hunk 1]
```diff
@@ -7,13 +7,17 @@
...
 typedef int errcode;
+#include "hiwire.h"
 
 int
 error_handling_init();
 
 errcode
 log_error(char* msg);
 
+errcode
```
Comment (dalcde) [inline src/core/error_handling.h:18]: Can you add a comment indicating that this is supposed to be not used in any existing code?

Comment (hoodmane) [reply src/core/error_handling.h:18]: You mean like something like "This is a debug tool, it's probably dead code but please don't remove it"?

Comment (dalcde) [reply src/core/error_handling.h:18]: Yeah. Explain that it is intentionally dead, not accidentally dea.

Comment (hoodmane) [reply src/core/error_handling.h:18]: Okay, will do.

**Review by dalcde — APPROVED** [review]

**Pull request merged**